lib/std/io/bit_reader.zig

```diff
@@ -9,7 +9,7 @@ const math = std.math;
 pub fn BitReader(comptime endian: std.builtin.Endian, comptime ReaderType: type) type {
     return struct {
         forward_reader: ReaderType,
-        bit_buffer: u7,
+        bit_buffer: ?u8,
         bit_count: u3,
```

Comment:
2**3-1 = 7 bit maximum, so I think this must become u4 for 8 bits.
Unless 0 is not used for 0 bit types, but then there must be always an 1 offset applied and that looks weird.

Reply:
Yep, I mentioned in the description there's a offset by one, since there's no need to represent 0 bits in the buffer now. I can do it with a u4 instead, but seemed a little neater to make the representation as tight as I could.